pkg/kubernetes/kubernetes_interface.gen.go

```diff
@@ -9,157 +9,232 @@ import (
-	GetConfigMap(ctx context.Context, namespace, name string) (*corev1.ConfigMap, error)
+	GetConfigMap(ctx context.Context, key ctrlclient.ObjectKey) (*corev1.ConfigMap, error)
+	// CreateConfigMap creates k8s configmap.
+	CreateConfigMap(ctx context.Context, config *corev1.ConfigMap) (*corev1.ConfigMap, error)
+	// UpdateConfigMap updates k8s configmap.
+	UpdateConfigMap(ctx context.Context, config *corev1.ConfigMap) (*corev1.ConfigMap, error)
+	// GetClusterServiceVersion retrieves a ClusterServiceVersion by namespaced name.
+	GetClusterServiceVersion(ctx context.Context, key ctrlclient.ObjectKey) (*olmv1alpha1.ClusterServiceVersion, error)
+	// ListClusterServiceVersion list all CSVs for the given namespace.
+	// This method returns a list of full objects (meta and spec).
+	ListClusterServiceVersion(ctx context.Context, opts ...ctrlclient.ListOption) (*olmv1alpha1.ClusterServiceVersionList, error)
+	// DeleteClusterServiceVersion deletes a ClusterServiceVersion.
+	DeleteClusterServiceVersion(ctx context.Context, obj *olmv1alpha1.ClusterServiceVersion) error
+	// DeleteClusterServiceVersions deletes all ClusterServiceVersion in the given namespace.
+	// This function will wait until all ClusterServiceVersion are deleted.
+	DeleteClusterServiceVersions(ctx context.Context, opts ...ctrlclient.ListOption) error
+	// ListCRDs lists all CRDs.
+	ListCRDs(ctx context.Context, opts ...ctrlclient.ListOption) (*apiextv1.CustomResourceDefinitionList, error)
+	// DeleteCRD deletes a CRD by name.
+	DeleteCRD(ctx context.Context, obj *apiextv1.CustomResourceDefinition) error
+	// ListDatabaseClusters returns list of managed database clusters in a given namespace.
+	// This method returns a list of full objects (meta and spec).
+	ListDatabaseClusters(ctx context.Context, opts ...ctrlclient.ListOption) (*everestv1alpha1.DatabaseClusterList, error)
+	// GetDatabaseCluster returns database clusters by provided name and namespace.
+	GetDatabaseCluster(ctx context.Context, key ctrlclient.ObjectKey) (*everestv1alpha1.DatabaseCluster, error)
+	// DeleteDatabaseCluster deletes database cluster by provided name and namespace.
+	DeleteDatabaseCluster(ctx context.Context, obj *everestv1alpha1.DatabaseCluster) error
+	// CreateDatabaseCluster creates database cluster.
+	CreateDatabaseCluster(ctx context.Context, cluster *everestv1alpha1.DatabaseCluster) (*everestv1alpha1.DatabaseCluster, error)
+	// UpdateDatabaseCluster updates database cluster.
+	UpdateDatabaseCluster(ctx context.Context, cluster *everestv1alpha1.DatabaseCluster) (*everestv1alpha1.DatabaseCluster, error)
+	// DeleteDatabaseClusters deletes all database clusters in provided namespace.
+	// This function will wait until all clusters are deleted.
+	DeleteDatabaseClusters(ctx context.Context, opts ...ctrlclient.ListOption) error
+	// DatabasesExist checks if there are databases that match criteria exist.
+	DatabasesExist(ctx context.Context, opts ...ctrlclient.ListOption) (bool, error)
```

Comment:
I'm not sure I'd expose the entire `ListOptions` for `DatabaseExists`. Maybe we can be more specific so that the caller needs to only pass a list of namespaces, and internally we can convert that to list options

Reply:
why not? in fact this method is a simplified version of `ListDatabases` but used when caller is not interested in details of existing databases. And it looks logical to have the same prototype for both